### ISSUE TITLE
Model update causes editor to lose focus

### ISSUE THREAD

**Issue by Eblax** [fraywing/textAngular#1458]
When there is an element showing the contents of the model with ng-bind-html, after each update to the model the caret disappears and you need to click inside the textAngular editor to continue typing.

- This seems to happen only if the display element is put earlier than the textAngular element in the DOM.
- This happens in all versions from 1.5.13 to 1.5.16. 
- This does not happen in version 1.5.12, it seems to have been introduced with version 1.5.13.

Plunkr showing the problem:
http://plnkr.co/edit/8JhHIO8PRhhtg53L0VTe?p=preview

How to reproduce: 
- Have a textAngular element on the page with an element bound to the same model before it:
`<div ng-bind-html="htmlcontent"></div>`
`<div text-angular="text-angular" ng-model="htmlcontent"></div>`

- Type something in the textAngular editor and then wait a second

- Notice the caret disappears and you can't continue typing

**Comment by srobledo** [comment]
same thing here, but if i downgrade to 1.5.12 @Eblax i get this error on IE 

`Unable to get property '0' of undefined or null reference`

**Comment by gbittmann** [comment]
I'm seeing the same behavior. 

Thanks for isolating the issue, a quick test of moving the ng-bind-html below the text-angular element and the issue disappears.

I would also like to add that while this same behavior is occurring I noticed lot of rangy selection spans get left in the model.

**Comment by MarkOdey** [comment]
I am having that problem too. 

After investigation, I only get the issue in the minified version. If I use the uncompressed version everything works as expected.

**Comment by bostondevin** [comment]
+1

**Comment by MarkOdey** [comment]
little discovery of late afternoon! This is being carried all around.

id="selectionBoundary_1489437873144_8958312861243121"

I belive that something is not being stripped properly during the edition of the text.

The version I am using is 1.5.16

Could this bug be related to : 

https://github.com/fraywing/textAngular/issues/1290

**Comment by Eblax** [comment]
Good catch! That does seem to be the issue. I see there are 3 regexps on line ~425 that are supposed to strip them out, but they fail because there is an inline style attribute on the spans. 

When I add .*? in the third regexp to match any extra characters until the end of the <span> tag, the problem disappears:
`var rsb3 = new RegExp(/<span id="selectionBoundary_\d+_\d+" class="rangySelectionBoundary".*?>[^<>]+?<\/span>/ig);`

Edit: Changed .+? to .*? to avoid breaking existing functionality

**Comment by MarkOdey** [comment]
In deed this seems to be solving the problem. Could we make it a pull request for next release? Eblax would you do it for us?

**Comment by gbittmann** [comment]
Using 1.5.16 I'm seeing this behavior even if the element with ng-bind-html after textangular. 

Not sure if the proposed fix covers this or not since the initial issue only had the problem occurring if ng-bind-html was before textangular

**Comment by gbittmann** [comment]
Additional details, ui.router is being used.

Content that is not inside of a view works as expected.
Content inside of a view is showing this behavior.

**Comment by gbittmann** [comment]
The MarkOdey solution fixes the issue for me as well.

**Comment by volmerf** [comment]
We're also having the problem referenced in #1290. It's still happening in 1.5.16.

**Comment by kanadbahalkar** [comment]
@Eblax is a savior!

**Comment by akve** [comment]
the issue still exists, and solved by the strange solution :) thanks @Eblax
